include/boost/astronomy/coordinate/spherical_representation.hpp

```diff
@@ -16,136 +21,447 @@
-                {
-                    boost::geometry::set<2>(this->point, distance);
-                }
-
-                boost::astronomy::coordinate::spherical_representation<DegreeOrRadian>
-                    operator +(boost::astronomy::coordinate::spherical_differential<DegreeOrRadian> const& diff) const
-                {
-                    boost::astronomy::coordinate::spherical_representation<DegreeOrRadian> temp(this->point);
-
-                    temp.set_lat(temp.get_lat() + diff.get_dlat());
-                    temp.set_lon(temp.get_lon() + diff.get_dlon());
-                    temp.set_dist(temp.get_dist() + diff.get_ddist());
-
-                    return temp;
-                }
-
-                boost::astronomy::coordinate::spherical_representation<DegreeOrRadian>
-                    operator +(boost::astronomy::coordinate::spherical_coslat_differential<DegreeOrRadian> const& diff) const
-                {
-                    boost::astronomy::coordinate::spherical_representation<DegreeOrRadian> temp(this->diff);
-
-                    temp.set_dlat(temp.get_lat() + diff.get_dlat());
-                    temp.set_dlon_coslat(temp.get_lon() + diff.get_dlon_coslat());
-                    temp.set_ddist(temp.get_dist() + diff.get_ddist());
-
-                    return temp;
-                }
-            };
-        }//namespace coordinate
-    } //namespace astronomy
-} //namespace boost
-#endif // !BOOST_ASTRONOMY_COORDINATE_SPHERICAL_REPRESENTATION_HPP
+        this->set_lat_lon_dist(lat, lon, distance);
+    }
+
+    //!constructs object from boost::geometry::model::point object
+    template
+    <
+        std::size_t OtherDimensionCount,
+        typename OtherCoordinateSystem,
+        typename OtherCoordinateType
+    >
+    spherical_representation
+    (
+        bg::model::point
+        <
+            OtherCoordinateType,
+            OtherDimensionCount,
+            OtherCoordinateSystem
+        > const& pointObject
+    )
+    {
+        bg::model::point<OtherCoordinateType, 3, bg::cs::cartesian> temp;
+        bg::transform(pointObject, temp);
+        bg::transform(temp, this->point);
+    }
+
+    //copy constructor
+    spherical_representation
+    (
+        spherical_representation
+        <
+            CoordinateType,
+            LatQuantity,
+            LonQuantity,
+            DistQuantity
+        > const& other)
+    {
+        this->point = other.get_point();
+    }
+
+    //!constructs object from any type of representation
+    template <typename Representation>
+    spherical_representation(Representation const& other)
+    {
+        BOOST_STATIC_ASSERT_MSG((boost::astronomy::detail::is_base_template_of
+            <boost::astronomy::coordinate::base_representation, Representation>::value),
+            "No constructor found with given argument type");
+
+        bg::model::point<typename Representation::type, 3, bg::cs::cartesian> temp;
```

Comment:
Here you should use `make` function as there can be same problem when cartesian is in different units

Reply:
You have to use `make_spherical_representation` instead of `cartesian`